nemo_skills/inference/generate.py

```diff
@@ -520,8 +508,25 @@ async def process_single_datapoint(self, data_point, all_data):
             if self.cfg.override_max_code_executions and self.cfg.total_code_executions_in_prompt is not None:
                 generation_params["max_code_executions"] = data_point["total_code_executions"]
 
+        # Tracking the tokens and generation time
+        input_sequence_length = None
+        if self.prompt is not None:
+            if generation_params["prompt"] is not None:
+                input_sequence_length = self.prompt.get_token_count(generation_params["prompt"])
+
+        # Start to track the generation time
+        start_time = time.time()
```

Comment:
actually could you move this outside of this function? Otherwise it's not going to work for cases when process_single_datapoint is overriden, which is quite common. We should just measure time for that method directly